#!/usr/bin/env python
 
PACKAGE = "xplane_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hold_roll", bool_t, 0, "Flag to test roll params", False)
gen.add("hold_pitch", bool_t, 0, "Flag to test pitch params", False)
gen.add("hold_course", bool_t, 0, "Flag to test course params", False)
gen.add("roll_step", double_t, 0 , "Commanded roll angle in degrees", 0.0, -90.0, 90.0)
gen.add("pitch_step", double_t, 0, "Commanded pitch angle in degrees", 0.0, -90.0, 90.0)
gen.add("Va_c", double_t, 0, "Commanded airspeed", 60.0, 0.0, 80.0)
gen.add("h_c", double_t, 0, "Commanded height", 50.0, 0.0, 1000.0)
gen.add("chi_c", double_t, 0, "Commanded course angle", 0.0, -180.0, 180.0)
gen.add("phi_ff", double_t, 0, "Commanded feedforward angle for orbits", 0.0,-180.0, 180.0)

exit(gen.generate(PACKAGE, "xplane_ros","Commands"))
